core/src/pixelcolor/transparent_color.rs

```diff
@@ -155,4 +156,183 @@ pub trait AlphaColor: PixelColor {
     /// Associated AlphaColor
     type AlphaColor: AlphaColor + ColorBlend<Self>;
+
+    /// Create transparent color from opaque color
+    fn with_alpha(self, alpha: u8) -> Self::AlphaColor;
+}
+
+macro_rules! argb_color {
+    (
+        $type:ident,
+        $base_type:ty,
+        $data_type:ty,
+        $storage_type:ty,Argb =
+        ($a_bits:expr, $r_bits:expr, $g_bits:expr, $b_bits:expr)
+    ) => {
+        impl_argb_color!(
+            $type,
+            $base_type,
+            $data_type,
+            $storage_type,
+            ($a_bits, $r_bits, $g_bits, $b_bits),
+            ($r_bits + $g_bits + $b_bits, $g_bits + $b_bits, $b_bits, 0)
+        );
+    };
+
+    (
+        $type:ident,
+        $base_type:ty,
+        $data_type:ty,
+        $storage_type:ty,Bgra =
+        ($a_bits:expr, $r_bits:expr, $g_bits:expr, $b_bits:expr)
+    ) => {
+        impl_argb_color!(
+            $type,
+            $base_type,
+            $data_type,
+            $storage_type,
+            ($a_bits, $r_bits, $g_bits, $b_bits),
+            (0, $r_bits, $r_bits + $g_bits, $r_bits + $g_bits + $a_bits)
+        );
+    };
 }
+
+macro_rules! impl_argb_color {
+    (
+        $type:ident,
+        $base_type:ty,
+        $data_type:ty,
+        $storage_type:ty,
+        ($a_bits:expr, $r_bits:expr, $g_bits:expr, $b_bits:expr),
+        ($a_pos:expr, $r_pos:expr, $g_pos:expr, $b_pos:expr)
+    ) => {
+        #[doc = "TODO"]
+        #[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Default, Debug)]
+        pub struct $type($storage_type);
+
+        impl $type {
+            // those are private in the base type so we have to recreate them here
+            const A_MASK: $storage_type = ($type::MAX_A as $storage_type) << $a_pos;
+            const R_MASK: $storage_type = ($type::MAX_R as $storage_type) << $r_pos;
+            const G_MASK: $storage_type = ($type::MAX_G as $storage_type) << $g_pos;
+            const B_MASK: $storage_type = ($type::MAX_B as $storage_type) << $b_pos;
+            //const RGB_MASK: $storage_type = Self::R_MASK | Self::B_MASK | Self::G_MASK;
+            const ARGB_MASK: $storage_type = Self::A_MASK | Self::R_MASK | Self::B_MASK | Self::G_MASK;
+
+            /// create from channels
+            pub const fn new(r: u8, g: u8, b: u8, a: u8) -> Self {
+                // into_storage is not const, it would allow removing some code here
+                let a_shifted = (a & Self::MAX_A) as $storage_type << $a_pos;
+                let r_shifted = (r & Self::MAX_R) as $storage_type << $r_pos;
+                let g_shifted = (g & Self::MAX_G) as $storage_type << $g_pos;
+                let b_shifted = (b & Self::MAX_B) as $storage_type << $b_pos;
+                Self(a_shifted | r_shifted | g_shifted | b_shifted)
+            }
+        }
+
+        impl PixelColor for $type {
+            type Raw = $data_type;
+        }
+
+        impl ColorBlend<$base_type> for $type {
```

Comment:
Do you want to add `impl ColorBlend<$type> for $type` in this PR or leave it for a future PR?